src/cholupdates/rank_1/_seeger.py

```diff
@@ -0,0 +1,262 @@
+"""Interface functions for the rank-1 up- and downdate algorithms specified in sections
+2 and 3 of [1].
+
+References
+----------
+.. [1] M. Seeger, "Low Rank Updates for the Cholesky Decomposition", 2008.
+"""
+
+from typing import Optional
+
+import numpy as np
+
+from ._arg_validation import _validate_update_args
+from ._seeger_impl_python import update as _update_impl_python
+
+_update_available_impls = ["python"]
+
+try:
+    from ._seeger_impl_cython import update as _update_impl_cython
+
+    _cython_available = True
+    _update_available_impls.append("cython")
+
+    _update_impl_default = _update_impl_cython
+except ImportError:
+    _cython_available = False
+
+    _update_impl_default = _update_impl_python
+
+
+def update_seeger(
```

Comment:
In the tech report, Matthias Seeger basically uses the implementation of these functions from LINPACK, but he modifies  them, because they can actually produce negative diagonal entries. So he does in a way modify the implementation, but not the methods. Do you think the comment should somehow reflect that?

Reply:
Also, this methods are specifically postfixed `*_seeger`, because it is a specific implementation of the rank-1 update. There could at some point be other methods which implement different algorithms.

Users who find this naming opaque should use the interface method `cholupdates.rank_1.update` anyway.

Reply:
A small note on how this algorithm differs from the one in LINPACK would be helpful in the docs, I think.

Reply:
Done